src/ai/OpenRouter.js

```diff
@@ -8,17 +8,35 @@ export const OpenRouter = class extends OpenAI {
   constructor(options) {
     options.baseURL ||= 'https://openrouter.ai/api/v1';
     super(options);
+    const models = this.model.split(';');
+    if (models.length > 1) {
+        this.model = models[0];
+        this.models = models.slice(1);
+    }
   }
 
   async init() {
     if (this.didInit) {
       return;
     }
 
-    const parts = this.model.split('/');
-    const data = await getModelData(parts[0], parts[1], this.cache);
+    // Get primary model information
+    let parts = this.model.split('/');
+    let data = await getModelData(parts[0], parts[1], this.cache);
     this.maxTokens = data.maxTokens;
     this.pricing = data.pricing;
+
+    // Get fallback model information
+    if (this.models) {
+      for (let fallbackModel of this.models) {
```

Comment:
Do them concurrently:

```
const datas = await Promise.all(this.fallbacks.map(it => {
   // your code here to do one
});
// do stuff with datas here
```

docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

note the difference between Promise.all() and Promise.allSettled():
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled